Show localized version of string resources.

lsoft opened the pull request:
I added controls to show localized versions of string resource, like this:

<img width="1036" height="600" alt="image" src="https://github.com/user-attachments/assets/fe760ed5-f5d2-48fb-bf78-8ea4ba4f7584" />


Do you accept such effort? If so, please test my changes, unfortunately, the changes are huge, and I'm not sure I did not break something.

Thanks for this extension!

lsoft committed:
Show localized version of string resources.

mrlacey commented:
- What was the reason for this change? What (as a user) is the scenario you're trying to support?
- What was wrong with the setting-based way of specifying the display of something other than the default language?
- In that you've removed a feature that other people were using, how should they adapt to this change? Will they have to always select their preferred language every time they open a project?
- Do the links need to be displayed in every adornment? I can imagine that being overwhelming and very cluttered in a lot of situations.
- What happens when lots of language translations are included? And/or when the displayed text is very long?

I'm open to contributions, but as with almost every other project I've encountered, it's best to discuss changes before submitting a PR.

Right now, I'd not be looking to accept this as I don't understand the reason for the change, or the consequences it has on existing users/behavior.

lsoft commented:
> What was the reason for this change? What (as a user) is the scenario you're trying to support?

Allow the programmer to see the string resources for any locale, without opening resx file.

> What was wrong with the setting-based way of specifying the display of something other than the default language?

In situ switch is better that something far away. It is just about comfort if the programmes needs to see localized strings.

> In that you've removed a feature that other people were using, how should they adapt to this change? Will they have to always select their preferred language every time they open a project?

Hmm.. If I understand the code right, I did not remove any feature. What the feature I removed accidentally? Current UI culture is used if preferred culture is empty, so there is a backward compatibility. I tried not to break anything.

> Do the links need to be displayed in every adornment? I can imagine that being overwhelming and very cluttered in a lot of situations.

For every, yes. Please, share your vision where these controls will be undesirable?

> What happens when lots of language translations are included?

Good point. PR needs to be improved in some way to process this situation.

> And/or when the displayed text is very long?

This vsix already have the guard about this; the guard is still here, so I did not change this behaviour. Am I wrong?

> it's best to discuss changes before submitting a PR

Yes, this is the better way. But I needed this feature, so I implemented it. PR is just a sharing values back. If you have any strong cons for this change, I'm fully OK with this. If you think PR needs to be improved additionally to be accepted, please let me know.

mrlacey commented:
So, it looks like your goal is to support developers who want to switch between displaying text for a small number of languages and changing what is shown for individual strings. I've not heard this requirement before, and so I do not expect it to be a common enough scenario to change the way the extension works.

In response to the points above (some of which further indicate the need for a rethink parts of what's been changed)



Having the switching links next to every string becomes overwhelming when there are lots of languages supported.
Also, note on the second line, the existing logic for detecting overlapping adornments is broken by the addition of these links.

<img width="1318" height="504" alt="image" src="https://github.com/user-attachments/assets/000543e1-6203-4ac1-80aa-98dc9ba49c8f" />


I was surprised that you don't load the neutral languages by default. This is a major change to the way the extension worked previously.

After loading them, I found the additions even more overwhelming and cluttering of the UI. The highlight also shows that you're not accounting for the links appearing over other text on the same line, which makes them unreadable.

<img width="1480" height="396" alt="image" src="https://github.com/user-attachments/assets/52f66c95-0f90-40a3-b238-7770ce08dcec" />


The current UI Culture doesn't always match the neutral language. Previously, the neutral language was shown as that's what most developers use and work with, but having an option for an alternative (the option/setting you removed) to be selected in preference supported others. It also served as a simple, single way to change all displayed strings at once.  This is the first time I've heard of a developer wanting to see a mix of languages in the editor at the same time. Comparing different translations of text within the context of the code (rather than as displayed in the UI) is an interesting scenario I've not encountered or considered before.





Wrapping is broken in your PR. The following shows an example with wrapping enabled: in the upper example, the clickable options go off the viewable area. The lower highlight shows adornments not being displayed on a wrapped line. These are breaks from the current release version.

<img width="1536" height="577" alt="image" src="https://github.com/user-attachments/assets/5af388cf-b5e1-4fa6-bc8a-ef9cd34bee64" />





I'm glad you were able to modify things to create a version that works the way you want, but I won't be taking your version forward as a replacement for what exists today.
